src/common/scheduler/watcher.go

```diff
@@ -0,0 +1,128 @@
+package scheduler
+
+import (
+	"github.com/vmware/harbor/src/common/scheduler/policy"
+	"github.com/vmware/harbor/src/common/scheduler/task"
+	"github.com/vmware/harbor/src/common/utils/log"
+)
+
+//Watcher is an asynchronous runner to provide an evaluation environment for the policy.
+type Watcher struct {
+	//The target policy.
+	p policy.Policy
+
+	//The channel for receive stop signal.
+	cmdChan chan bool
+
+	//Indicate whether the watch is started and running.
+	isRunning bool
+
+	//Report stats to scheduler.
+	stats chan *StatItem
+
+	//If policy is automatically completed, report the policy to scheduler.
+	doneChan chan string
+}
+
+//NewWatcher is used as a constructor.
+func NewWatcher(p policy.Policy, st chan *StatItem, done chan string) *Watcher {
+	return &Watcher{
+		p:         p,
+		cmdChan:   make(chan bool),
+		isRunning: false,
+		stats:     st,
+		doneChan:  done,
+	}
+}
+
+//Start the running.
+func (wc *Watcher) Start() {
+	if wc.isRunning {
+		return
+	}
+
+	if wc.p == nil {
+		return
```

Comment:
No, this is the start method, the wc should not be running, it's ready to run. Then we check if it bound policy is valid.